org.codeberg.LibreMenuEditor.yml

```diff
@@ -0,0 +1,42 @@
+app-id: org.codeberg.LibreMenuEditor
+runtime: org.gnome.Platform
+runtime-version: "44"
+sdk: org.gnome.Sdk
+command: libre-menu-editor
+finish-args:
+  - --env=FLATPAK_APP_DIR=/app/libre-menu-editor
+    # to know when app is running as flatpak
+    # to know where files are stored
+  - --share=ipc
+    # to support x11-fallback
+  - --device=dri
+    # to avoid problems with graphics
+  - --socket=wayland
+    # to provide wayland support
+  - --socket=fallback-x11
+    # to support x11 if wayland is not available
+  - --filesystem=xdg-data/applications
+    # for reading and writing .desktop files in .local/share/applications
+  - --filesystem=xdg-data/flatpak:ro
+    # for reading .desktop files in .local/share/flatpak/exports/share/applications
+    # to avoid unreachable symlink targets in .local/share/flatpak
+  - --filesystem=/var/lib/flatpak:ro
+    # for reading .desktop files in /var/lib/flatpak/exports/share/applications
+    # to avoid unreachable symlink targets in /var/lib/flatpak
+  - --filesystem=/var/lib/snapd:ro
+    # for reading .desktop files in /var/lib/snapd/desktop/applications
+    # to avoid unreachable symlink targets in /var/lib/snapd
+  - --filesystem=host:ro
+    # for reading .desktop files in /usr/local/share/applications
+    # for reading .desktop files in /usr/share/applications
+    # to avoid unreachable symlink targets
```

Comment:
this filesystem permissions are very excessive, and most of them not allowed by default.

Reply:
--filesystem=host:ro is necessary for two reasons:
  1. it can not reach /usr/share/applications, so I have to get it through /run/host
  2. if there is a symlink in any of the directories and it points to a directory that is not accessible, my program can not load the .desktop file

I had to include --filesystem=/var/lib/flatpak:ro because it wouldn't find /run/host/var/lib/flatpak:ro 
I had to include --filesystem=/var/lib/snapd:ro because it wouldn't find /run/host/var/lib/snapd:ro 
--filesystem=xdg-data/flatpak:ro is for loading .desktop files of user-installed flatpaks

--filesystem=xdg-data/applications needs write access, because it needs to write the users changes